packages/vertexai/src/requests/request-helpers.ts

```diff
@@ -49,11 +52,13 @@ export function formatNewContent(
   if (typeof request === 'string') {
     newParts = [{ text: request }];
   } else {
-    for (const partOrString of request) {
-      if (typeof partOrString === 'string') {
-        newParts.push({ text: partOrString });
+    for (const elem of request) {
+      // This throws an error if request is not iterable
```

Comment:
Is that bad? Does it throw an error that's not helpful to the user? Are the other incomplete comments in this function related to this problem?

Reply:
Sorry the comments here were stragglers from when I was just going through the SDK and trying to understand how things work. Removed them all